master/internal/configpolicy/task_config_policy.go

```diff
@@ -28,6 +30,65 @@ type NTSCConfigPolicies struct {
 	Constraints     *model.Constraints   `json:"constraints"`
 }
 
+// ValidateNTSCConstraints returns true if the NTSC config passes constraint checks.
+func ValidateNTSCConstraints(
+	ctx context.Context,
+	workspaceID int,
+	workloadConfig model.CommandConfig,
+	resourceManager rm.ResourceManager,
+) (bool, error) {
+	constraints, err := GetMergedConstraints(ctx, workspaceID, model.NTSCType)
+	if err != nil {
+		return false, err
+	}
+
+	// For each submitted constraint, check if the workload config is within allowed values.
+	smallerHigher, err := resourceManager.SmallerValueIsHigherPriority()
+	if err == nil && constraints.PriorityLimit != nil && workloadConfig.Resources.Priority != nil {
```

Comment:
Maybe a different approach would be better?

We have an RM interface function to find out if an RM uses `smaller value -> higher priority`, `larger value -> higher priority`, or `does not implement priority`. Here's the implementation for dispatcherrm: https://github.com/determined-ai/determined/blob/3ac85df3d5a1c2fbfc0f5f0c1f362eb8b2c58bb2/master/internal/rm/dispatcherrm/dispatcher_resource_manager.go#L2261-L2264

It would probably be best to return an error if an admin attempts to set a priority constraint for a dispatcherrm cluster ... but I would prefer not to require that the API layer know about the RM. 

Definitely can add a comment to make this more clear.